### PR TITLE
Enhanced model validation

### PR DESCRIPTION
**Pull request by radowit**
https://docs.google.com/spreadsheets/d/11W7-t7uITbiP66T7R90mOU01-dp4EsbeZIOfWkrEWio/edit?usp=sharing

These are some big changes in django models that hopefully won't stop our site :)

## Automatic setting SlugField in Django Admin

All SlugFields are required now and if you editing them in Django Admin the slug is computed dynamically
Also, WaitList and Room have now their own slug field.

## Dynamic status fields

Added to a couple of models and set automatically when certain conditions are met.
This will help us to set this logic in backend (and db), so that it won't be duplicated.
Also, all datetime fields in Festival are now not required. We are ok with draft festivals we don't need to show them in frontend now).

## Unique constraints

I set some constraints that make sense for me. Please look into them and verify if they are needed.

Ignore files added in `stubs` directory - thay are needed for type hints to work.

### CODE REVIEW

File: app/chronology/models.py [hunk 1]
```diff
@@ -17,41 +20,114 @@ def default_festival_settings() -> Dict[str, EmptyDict]:
...
+                    start_time__lt=F("end_time"),
+                ),
+                name="festival_date_times",
+            ),
+        ]
 
     def __str__(self) -> str:
         return self.name
 
+    @computed(
```
Comment (lprasol) [inline app/chronology/models.py:89]: Impressive, but could it be just `@property`?

File: app/chronology/models.py [hunk 2]
```diff
@@ -111,25 +210,46 @@ class AgendaItem(models.Model):
...
-        Room, on_delete=models.SET_NULL, blank=True, null=True, verbose_name=_("room")
+        Room, null=True, blank=True, on_delete=models.CASCADE, verbose_name=_("room")
     )
 
     class Meta:
         db_table = "ch_agenda_item"
         verbose_name = _("agenda item")
         verbose_name_plural = _("agenda items")
 
+    @computed(
```
Comment (lprasol) [inline app/chronology/models.py:221]: Same question as above

File: app/crowd/models.py [hunk 3]
```diff
@@ -20,6 +20,13 @@ class Meta:  # noqa: D106
         verbose_name = _("user")
         verbose_name_plural = _("users")
         db_table = "cr_user"
+        constraints = [
+            models.UniqueConstraint(
+                fields=["auth0_id"],
+                condition=~models.Q(auth0_id=""),
```
Comment (lprasol) [inline app/crowd/models.py:26]: Almost missed `~`, and it looked like unique empty string XD

File: app/locale/pl/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -24,181 +24,220 @@ msgstr ""
...
-#: app/notice_board/models.py:60
+#: app/chronology/models.py:149 app/crowd/models.py:20
+#: app/notice_board/models.py:66
 msgid "user"
 msgstr "użytkownik"
 
-#: app/chronology/models.py:85 app/chronology/models.py:94
+#: app/chronology/models.py:153 app/chronology/models.py:180
 msgid "helper"
 msgstr ""
```
Comment (lprasol) [inline app/locale/pl/LC_MESSAGES/django.po:125]: misssing?

File: app/chronology/models.py (hunk 1 above)
Comment (radowit) [reply app/chronology/models.py:89]: It won't be save in database then

File: app/chronology/models.py (hunk 2 above)
Comment (radowit) [reply app/chronology/models.py:221]: Same answer as above :)

File: app/crowd/models.py (hunk 3 above)
Comment (radowit) [reply app/crowd/models.py:26]: We want to ensure non-empty auth0 ids are unique

**Pull request merged**